addon/models/resource.js

```diff
@@ -415,8 +418,14 @@ Resource.reopenClass({
     for (let i = 0; i < attrs.length; i++) {
       prototype[attrs[i]] = {};
     }
+
     const instance = this._super(prototype);
     if (properties) {
+      if (properties.hasOwnProperty('id')) {
+        // JSONAPI ids MUST be strings.
+        properties.id = properties.id.toString();
```

Comment:
maybe it's worth adding an Assert in here, https://github.com/pixelhandler/ember-jsonapi-resources/blob/master/addon/serializers/application.js#L208-L233  ?